core/Pos/Core/Context.hs

```diff
@@ -9,9 +24,57 @@ module Pos.Core.Context
 
 import           Universum
 
-import           Pos.Core.Address (addressHash, makePubKeyAddress)
-import           Pos.Core.Types   (Address, StakeholderId)
-import           Pos.Crypto       (PublicKey, SecretKey, toPublic)
+import           Control.Lens       (makeLenses)
+import           Data.Reflection    (Given (..), give)
+
+import           Pos.Core.Address   (addressHash, makePubKeyAddress)
+import           Pos.Core.Types     (Address, BlockCount (..), StakeholderId, SlotCount)
+import           Pos.Crypto         (PublicKey, SecretKey, toPublic)
+import           Pos.Core.Constants (staticBlkSecurityParam)
+
+-- | Core constants. They should be really constant and never change.
+data CoreConstants = CoreConstants
+    { _ccBlkSecurityParam :: !BlockCount
+    }
+
+makeLenses ''CoreConstants
+
+type HasCoreConstants = Given CoreConstants
+
+giveConsts :: BlockCount -> (HasCoreConstants => r) -> r
```

Comment:
> we are using mkSafeConstructor... everywhere, it's yet another such place.

We use it only for types with internal invariants. We don't have any invariants for core constants.

I can write
```
myConstants = CoreConstants { _ccBlkSecurityParam = 5, _ccAnotherNumber = 10 }
```
and even if someone changes the order of this values in `CoreConstants`, the code will remain correct.